### PR TITLE
Use object destructuring for arguments

### PR DESCRIPTION
**Pull request by jaas666**
Here's my attempt to solve #10.

### CODE REVIEW

File: files/js/cookies.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
 /**
  * Set cookie.
- * 
- * @param {string} name Cookie name.
- * @param {string|boolean} value Cookie value. Defaults to boolean true.
+ * @param {Object} cookie setCookie function requires an object with the following properties:
+ * @param {string} cookie.name Cookie name.
+ * @param {string|boolean=} cookie.value Cookie value. Defaults to boolean true.
```
Comment (nikkifurls) [inline files/js/cookies.ts:5]:
```suggestion
 * @param {string|boolean} cookie.value Cookie value. Defaults to boolean true.
```
Typo :)

File: files/js/cookies.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 /**
  * Get cookie by name.
- * 
- * @param {string} name Cookie name.
+ * @param {Object} cookie getCookie function requires an object with the following properties:
+ * @param {string} cookie.name Cookie name.
  * @returns {string} Cookie value.
  */
-export const getCookie = (name: string): string => {
+export const getCookie = ({name}:{name: string}): string => {
```
Comment (nikkifurls) [inline files/js/cookies.ts:19]:
```suggestion
export const getCookie = ({ name }: { name: string }): string => {
```
I prefer spaces around the property names like you had in `setCookie()`

File: files/js/notifications.ts [hunk 3]
```diff
@@ -97,19 +98,19 @@ export const closeNotification = (): void => {
...
- * @param {string} customIconSelector Selector for the promo icon, if one exists. If provided, the icon will be animated and a cookie will be set on click so that it only animates once.
+ * @param {Object} notification The `showPromo` function requires an object with the following properties:
+ * @param {string} notification.text Text to display in notification.
+ * @param {string=} notification.cookieName Cookie name to provide to showNotification(), so that the promo does not display again for the user. Defaults to the website URL with '-promo' appended.
+ * @param {string=} notification.customIconSelector Selector for the promo icon, if one exists. If provided, the icon will be animated and a cookie will be set on click so that it only animates once.
  */
-export const showPromo = (text: string, cookieName: string = `${window.baseUrl}-promo`, customIconSelector: string = 'nav .custom'): void => {
+export const showPromo = ({text, cookieName = `${window.baseUrl}-promo`, customIconSelector = 'nav .custom'}:{text: string, cookieName?: string, customIconSelector?: string}): void => {
```
Comment (nikkifurls) [inline files/js/notifications.ts:113]:
```suggestion
export const showPromo = ({ text, cookieName = `${window.baseUrl}-promo`, customIconSelector = 'nav .custom' }: { text: string, cookieName?: string, customIconSelector?: string }): void => {
```

File: files/js/notifications.ts [hunk 4]
```diff
@@ -126,13 +127,13 @@ export const showPromo = (text: string, cookieName: string = `${window.baseUrl}-
 
 			element.classList.add('animate');
 			element.addEventListener('click', () => {
-				setCookie(`${cookieName}-icon`, true);
+				setCookie({name: `${cookieName}-icon`, value: true});
```
Comment (nikkifurls) [inline files/js/notifications.ts:130]:
```suggestion
				setCookie({ name: `${cookieName}-icon`, value: true });
```

File: files/js/notifications.ts [hunk 5]
```diff
@@ -126,13 +127,13 @@ export const showPromo = (text: string, cookieName: string = `${window.baseUrl}-
...
 			});
 		}, 20000);
 	}
 
 	// Show the notification for the promo.
 	setTimeout(() => {
-		showNotification(text, cookieName);
+		showNotification({text: text, cookieName: cookieName});
```
Comment (nikkifurls) [inline files/js/notifications.ts:137]:
```suggestion
		showNotification({ text: text, cookieName: cookieName });
```

File: files/js/urls.ts [hunk 6]
```diff
@@ -40,24 +40,23 @@ export const copyUrl = (url: string = window.location.href): void => {
...
-			`<a class='button facebook' href='https://www.facebook.com/sharer/sharer.php?u=${url}' title='Share on Facebook'><i class='fas fa-share-alt'></i>&nbsp;&nbsp;Facebook</a>` +
+		showNotification({
+      text: 			`<a class='button facebook' href='https://www.facebook.com/sharer/sharer.php?u=${url}' title='Share on Facebook'><i class='fas fa-share-alt'></i>&nbsp;&nbsp;Facebook</a>` +
 			`<a class='button twitter' href='https://twitter.com/intent/tweet?text=${textEncoded}' title='Share on Twitter'><i class='fas fa-share-alt'></i>&nbsp;&nbsp;Twitter</a>` +
 			`<a class='button email' href='mailto:?subject=Check+out+${title}!&body=${textEncoded}' title='Share on Email'><i class='fas fa-share-alt'></i>&nbsp;&nbsp;Email</a>` +
 			`<a class='button copy' href='#' title='Copy' data-url='${url}'><i class='fas fa-copy'></i>&nbsp;&nbsp;Copy</button>`,
-			'',
-			'notification-share'
-		);
+      className: 'notification-share'
```
Comment (nikkifurls) [inline files/js/urls.ts:58]:
```suggestion
      cookieName: '',
      className: 'notification-share'
```
We accidentally left out `cookieName` here :)

**Comment by nikkifurls** [comment]
This looks fantastic @jaas666! Thank you so much for contributing! 😃  

I left a few minor suggestions, mostly around consistent spacing. Would you mind reviewing those and making those changes?

Also, there are a couple of changes to the calls to `showNotification()` in https://github.com/nikkifurls/pwagenerator/blob/master/files/js/index.ts and https://github.com/nikkifurls/pwagenerator/blob/master/template-parts/head.php#L337 that we'll need to update. Would you want to make those updates?

Thank you again! ❤️

**Comment by jaas666** [comment]
Should look better. Thanks for the review.

**Comment by nikkifurls** [comment]
> Should look better. Thanks for the review.

Wonderful! This looks perfect. Thank you so much for your contribution! Please feel free to contribute more! 😃

**Pull request merged**